libs/components/forms/src/lib/modules/file-attachment/file-drop/file-drop.component.spec.ts

```diff
@@ -1361,3 +1379,114 @@ describe('File drop component', () => {
     helpController.expectCurrentHelpKey('helpKey.html');
   });
 });
+
+describe('File drop reactive component', () => {
+  let fixture: ComponentFixture<ReactiveFileDropTestComponent>;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [SkyFileDropModule],
+      providers: [provideSkyFileReaderTesting()],
+    });
+    fixture = TestBed.createComponent(ReactiveFileDropTestComponent);
+    fixture.detectChanges();
+  });
+
+  it('should mark control as touched on file `drop` event', () => {
+    expect(fixture.componentInstance.fileDrop.touched).toBeFalse();
+    const dropEl = fixture.debugElement.query(By.css('.sky-file-drop'));
+
+    const dropEvent = {
+      dataTransfer: {},
+      stopPropagation: function (): void {},
+      preventDefault: function (): void {},
+    };
+
+    dropEl.triggerEventHandler('drop', dropEvent);
+    expect(fixture.componentInstance.fileDrop.touched).toBeTrue();
+  });
+
+  it('should mark control as touched on file drop clicked', () => {
+    expect(fixture.componentInstance.fileDrop.touched).toBeFalse();
+    const dropEl = fixture.nativeElement.querySelector('.sky-file-drop');
+
+    dropEl.click();
+    fixture.detectChanges();
+
+    expect(fixture.componentInstance.fileDrop.touched).toBeTrue();
+  });
+
+  it('should mark control as touched on link added', () => {
+    expect(fixture.componentInstance.fileDrop.touched).toBeFalse();
+    const linkButton = fixture.debugElement.query(
+      By.css('.sky-file-drop-link button'),
+    );
+    const linkEl = fixture.debugElement.query(
+      By.css('.sky-file-drop-link input'),
+    );
+
+    linkEl.triggerEventHandler('input', { target: { value: 'link.url' } });
+    fixture.detectChanges();
+
+    linkButton.nativeElement.click();
+    fixture.detectChanges();
+
+    expect(fixture.componentInstance.fileDrop.touched).toBeTrue();
+  });
+
+  it('should mark control as touched on link blur', () => {
+    expect(fixture.componentInstance.fileDrop.touched).toBeFalse();
+    const linkEl = fixture.nativeElement.querySelector(
+      '.sky-file-drop-link input',
+    );
+
+    SkyAppTestUtility.fireDomEvent(linkEl, 'blur');
+    fixture.detectChanges();
+
+    expect(fixture.componentInstance.fileDrop.touched).toBeTrue();
+  });
+
+  it('should set file drop to required using form control', () => {
```

Comment:
Is the file drop marked "required" when the `required` attribute is added to the component's tag, too?

Reply:
yes there are unit tests covering that and checking that appropriate classes and aria labels are added. those were there before control value accessor was added